modules/s2i/bash/artifacts/usr/local/s2i/assemble

```diff
@@ -54,3 +59,10 @@ if [ "$S2I_ENABLE_JLINK" = "true" ]; then
     }
 
 fi
+
+if [ "$S2I_DELETE_SOURCE" == "true" ]; then
+  if [ -n "$(find $S2I_JLINK_TEMP_PATH -maxdepth 0 -type d ! -empty 2> /dev/null)" ]; then
```

Comment:
I think this can go, and just unconditionally call the `rm -rf` if S2I_DELETE_SOURCE == true.